packages/amplify_authenticator/example/integration_test/sign_up_with_username_test.dart

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// This test follows the Amplify UI feature "sign-in-with-username"
+// https://github.com/aws-amplify/amplify-ui/blob/main/packages/e2e/features/ui/components/authenticator/sign-up-with-username.feature
+
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+import 'package:amplify_authenticator/src/keys.dart';
+import 'package:amplify_authenticator/src/screens/authenticator_screen.dart';
+import 'package:amplify_authenticator/src/state/inherited_auth_viewmodel.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'config.dart';
+import 'utils/mock_data.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  final authenticator = MaterialApp(
+    home: Authenticator(child: Container()),
+  );
+
+  group('sign-in-with-username', () {
+    // Given I'm running the example "ui/components/authenticator/sign-up-with-username"
+    setUpAll(() async {
+      await loadConfiguration(
+        'ui/components/authenticator/sign-up-with-username',
+      );
+    });
+
+    // Scenario: Login mechanism set to "username"
+    testWidgets('Login mechanism set to "username"', (tester) async {
+      await tester.loadAuthenticator(authenticator);
+      await tester.navigateToSignUp();
+
+      // Then I see "Username" as an input field
+      expect(
+        // Match on the title of the username field since it is dependent on
+        // the username alias selected.
+        find.descendant(
+          of: find.byKey(keyUsernameSignUpFormField),
+          matching: find.text('Username'),
+        ),
+        findsOneWidget,
+      );
+    });
+
+    // Scenario: "Preferred Username" is included from `aws_cognito_signup_attributes`
+    testWidgets(
+        '"Preferred Username" is included from `aws_cognito_signup_attributes`',
+        (tester) async {
+      // TODO: Clarify requirements
```

Comment:
Is this TODO from JS? I think there may have been some confusion from the `alias` change, but in this case it is valid to just include this as a regular signup attribute if it's in the config.

Reply:
No, it's for us to clarify - I agree, they probably just need to update their tests. But with the zero-config auth change, preferred username would not be shown in addition to the username field if the user selects `Username` as their login mechanism.

Reply:
If that's the case right now, it may be a bug. I think that preferred_username is a valid attribute to collect in addition to username/phone/email. The use case would be a *mutable* display name in a social media app, or something of that nature. I'll clarify this.

Reply:
Sounds good, thanks